addons/print/models/print_model.py

```diff
@@ -0,0 +1,295 @@
+# -*- coding: utf-8 -*-
+import base64
+import re
+import time
+
+from openerp import api, fields, models
+from openerp.tools.safe_eval import safe_eval
+from openerp.tools.translate import _
+
+
+class print_provider(models.Model):
+    """ Print Provider Base Model. Each specific provider can extend the model by adding
+        its own fields, using the provider_name (provider field) as a prefix for the new
+        fields and method.
+    """
+
+    _name = 'print.provider'
+    _rec_name = 'name'
+    _description = 'Print Provider'
+
+    def _get_providers(self):
+        return []
+
+    # --------------------------------------------------
+    # MODEL FIELDS
+    # --------------------------------------------------
+    name = fields.Char("Name", required=True)
+    environment = fields.Selection([('test', 'Test'), ('production', 'Production')], "Environment", default='test')
+    provider = fields.Selection(selection='_get_providers', string='Provider', required=True)
+    balance = fields.Float("Credit", digits=(16, 2))
+
+    # --------------------------------------------------
+    # METHOD to be redefined in the provider implemenation
+    # --------------------------------------------------
+    @api.multi
+    def update_account_data(self):
+        """ Update the provider account data. Requires a fetch to the provider server. """
+        self.ensure_one()
+        if hasattr(self, '%s_update_account_data' % self.provider):
+            return getattr(self, '%s_update_account_data' % self.provider)()
```

Comment:
Should the method return something ? Because there is a mismatch (method defined -> returns something, no method defined - >returns nothing). If there is a return data planned, it should be specified in the method's comment.

As this is just to update the account, I don't think it has to return something.